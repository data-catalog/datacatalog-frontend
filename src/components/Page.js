import React from 'react';
import { Container } from 'react-bootstrap';
import NotificationsSystem, { wyboTheme, useNotifications } from 'reapop';
import Appbar from './Appbar';
import AuthModal from './AuthModal';

const Page = ({ children }) => {
  const { notifications, dismissNotification } = useNotifications();
  return (
    <>
      <NotificationsSystem
        notifications={notifications}
        dismissNotification={(id) => dismissNotification(id)}
        theme={wyboTheme}
      />
      <AuthModal />
      <Appbar />
      <Container className="my-5">{children}</Container>
    </>
  );
};

export default Page;
