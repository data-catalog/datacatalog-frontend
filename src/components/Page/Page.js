import React from 'react';
import styled, { ThemeProvider } from 'styled-components';
import Navbar from '../Navbar';
import Sidebar from '../Sidebar';
import { Colors } from '../Global/Colors';

const WindowContainer = styled.div`
  position: absolute;
  top: 97px;
  right: 0;
  bottom: 0;
  left: 0;

  display: flex;
`;

const MainSideContainer = styled.div`
  background-color: ${(props) => props.theme.back_color};
  padding: 2% 2% 2% 2%;
  flex: 1;
  display: flex;
  overflow: visible;
  flex-direction: column;
`;

const Page = ({ children }) => {
  return (
    <>
      <Navbar />

      <ThemeProvider theme={Colors}>
        <WindowContainer>
          <Sidebar />
          <MainSideContainer>{children}</MainSideContainer>
        </WindowContainer>
      </ThemeProvider>
    </>
  );
};

export default Page;
