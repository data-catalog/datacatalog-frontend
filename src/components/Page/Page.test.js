import React from 'react';
import { shallow } from 'enzyme';
import Page from './Page';

describe('Page', () => {
  it('renders correctly', () => {
    const tree = shallow(<Page />);
    expect(tree).toMatchSnapshot();
  });

  it('renders its children', () => {
    const element = <p>children here</p>;
    const wrapper = shallow(<Page>{element}</Page>);

    expect(wrapper.contains(element)).toBe(true);
  });
});
