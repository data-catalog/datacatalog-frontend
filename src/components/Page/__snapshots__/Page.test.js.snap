// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page renders correctly 1`] = `
<Fragment>
  <Navbar />
  <Component
    theme={
      Object {
        "back_color": "#095E6D",
        "button_darker": "#0092A6",
        "detail_color": "#59858C",
        "font_color": "#FFFFFF",
        "header_color": "#05333B",
        "input_error": "rgba(255, 0, 0, 0.8)",
        "lightblue": "#00B4CC",
        "sidebar_color": "#007382",
        "white": "#FFFFFF",
      }
    }
  >
    <styled.div>
      <Sidebar />
      <styled.div />
    </styled.div>
  </Component>
</Fragment>
`;
